app/controllers/early_allocations_controller.rb

```diff
@@ -5,44 +5,48 @@ class EarlyAllocationsController < PrisonsApplicationController
 
   def new
     case_info = CaseInformation.find_by offender_id_from_url
-    @early_assignment = case_info.early_allocations.new
+    @early_assignment = EarlyAllocationStage1.new offender_id_from_url
     if case_info.local_divisional_unit.try(:email_address)
       render
     else
       render 'dead_end'
     end
   end
 
-  def create
-    @early_assignment = EarlyAllocation.new early_allocation_params.merge(offender_id_from_url)
-    if @early_assignment.save
+  def stage1
+    form_params = stage1_params.merge(offender_id_from_url)
+    form = EarlyAllocationStage1.new form_params
+    if form.valid?
+      @early_assignment = EarlyAllocation.new form_params.merge(default_params)
       if @early_assignment.eligible?
+        @early_assignment.save!
         if @offender.within_early_allocation_window?
           AutoEarlyAllocationEmailJob.perform_later(@prison.code, @offender.offender_no, Base64.encode64(pdf_as_string))
         end
         render 'eligible'
       else
-        render 'ineligible'
+        @early_assignment = EarlyAllocationStage2.new form_params
+        render 'stage2'
       end
     else
-      @early_assignment.errors.delete(:stage2_validation)
-      render create_error_page
+      @early_assignment = form
+      render 'new'
     end
   end
 
-  # record a community decision (changing 'maybe' into a yes or a no)
-  # can only be performed on the last early allocation record
-  def edit
-    @early_assignment = EarlyAllocation.where(offender_id_from_url).last
-  end
-
-  def update
-    @early_assignment = EarlyAllocation.where(offender_id_from_url).last
-
-    if @early_assignment.update(community_decision_params)
-      redirect_to prison_prisoner_path(@prison.code, @early_assignment.nomis_offender_id)
+  def create
```

Comment:
It would be nice if this method was called `stage2` – in the same way that there's a `stage1` method.

The same is true for the discretionary route – would be nice to have a `stage3` method and form object. Right now we have the `discretionary` method, but no 'stage 3' form object.

This approach would give us a really clear & obvious relationship between the view => form object => controller method for each stage of the form, and sets a strong convention which can be followed elsewhere in the future.

It would mean losing the `create` method altogether – which I realise breaks the traditional REST resource methods. But since this is a multi-stage form/wizard, is there really such thing as one `create` endpoint? The creation could happen in any one of the stages (e.g. in stage 1 as 'eligible'), so it could be argued that the convention of having one single `create` method isn't helpful here.